File: SARAssist/AreaModel.m
//
//  AreaModel.m
//  SARAssist
//
//  Created by V-FEXrt on 7/20/15.
//  Copyright (c) 2015 V-FEXrt. All rights reserved.
//

#import "AreaModel.h"

@implementation AreaModel

@dynamic Name;
@dynamic Location;
@dynamic IsComplete;
@dynamic NorthEastLat;
@dynamic NorthEastLng;
@dynamic SouthWestLat;
@dynamic SouthWestLng;
@dynamic SearchAreaID;

+(NSString *)parseClassName{
    return @"SearchArea";
}

+(void)load{
    [AreaModel registerSubclass];
}

/*
-(void)setIsCompleteTo:(BOOL)complete{
    self[@"IsComplete"] = [NSNumber numberWithBool:complete];
}

-(CLLocationCoordinate2D)CLLoction{
    return CLLocationCoordinate2DMake([[self location] latitude], [[self location]longitude]);
}
*/

@end

File: SARAssist_Swift/SARAssist/AreaModel.swift
//
//  AreaModel.swift
//  SARAssist
//
//  Created by V-FEXrt on 9/18/15.
//  Copyright (c) 2015 V-FEXrt. All rights reserved.
//

import UIKit
import Parse

class AreaModel: PFObject, PFSubclassing {
    
    override internal class func initialize() {
        registerSubclass()
    }
    
    internal class func parseClassName() -> String {
        return "SearchArea"
    }
    
    @NSManaged internal var Name: String!
    @NSManaged internal var SearchAreaID: String!
    
    @NSManaged internal var Location: PFGeoPoint!
    @NSManaged internal var NorthEastLat: NSNumber!
    @NSManaged internal var NorthEastLng: NSNumber!
    @NSManaged internal var SouthWestLat: NSNumber!
    @NSManaged internal var SouthWestLng: NSNumber!
    

    //Special case for Bool values
    /*var IsComplete: Bool {
        get { return self["IsComplete"] as! Bool }
        set { self["IsComplete"] = newValue }
    }*/


}
